### PR TITLE
[NOT READY] GAGS 1 of 2: Framework

### PR DESCRIPTION
**Pull request by Nobelium-XVIII**
## About The Pull Request
From original PR:

"We've gotten occasional greyscale sprites and code throughout the years but they usually consist of a single sprite that gets colored with the color var. It doesn't look as good as it could and only allows for certain kinds of recolors. This system is to change that, it's a standardized way to make and configure greyscale sprites consisting of multiple layers.

## Info for spriters
There's a readme in code/datums/greyscale/ that has an overview and some basic instructions. It requires minimal code, and a debug tool is available to help get the colors and layers all working together.

The debug tool in action https://puu.sh/HuF2s/b406b4066c.gif
## Info about the back end
Please ask about any parts of the system that you don't understand, it's designed to be usable by non coders.

Partly due to overlay spam being a problem for clients, but also because it seemed to be the best way to have greyscale sprites without other code needing to be aware of it; Sprites are created with icon operations then cached for future use. The same greyscale config with the same color arguments will only ever make that sprite once. The performance implications of this are hard to measure but luckily this approach can easily be made to do most of its processing in init.

Generation of layers in this system has been abstracted to just being a collection of objects returning an icon. This means you could actually pull some fancy effects with custom layers if that's wanted in the future.

## Why It's Good For The Game
This should allow us to make use of greyscale images more often without them being burdened by lots of code or poor quality recolors. Once this is stable certain kinds of sprites should be mandatory greyscale moving forward, mainly any sprite where many recolors are an important aspect."

## Changelog
:cl:
refactor: Refactors greyscale sprite generation
admin: Added GAGS Admin tooling
/:cl:

### CODE REVIEW

**Comment by triplezeta** [comment]
convert my amoung pequenos to gags

**Comment by Nobelium-XVIII** [comment]
> convert my amoung pequenos to gags

later

**Comment by Nobelium-XVIII** [comment]
fuck

File: code/game/atoms.dm [hunk 1]
```diff
@@ -1062,6 +1086,8 @@
 	VV_DROPDOWN_OPTION(VV_HK_TRIGGER_EXPLOSION, "Explosion")
 	VV_DROPDOWN_OPTION(VV_HK_RADIATE, "Radiate")
 	VV_DROPDOWN_OPTION(VV_HK_EDIT_FILTERS, "Edit Filters")
+	if(greyscale_colors)
+		VV_DROPDOWN_OPTION(VV_HK_MODIFY_GREYSCALE, "Modify greyscale colors"
```
Comment (MarkSuckerberg) [inline code/game/atoms.dm]:
```suggestion
		VV_DROPDOWN_OPTION(VV_HK_MODIFY_GREYSCALE, "Modify greyscale colors")
```

**Comment by rye-rice** [comment]
this wont be good for the spriter economy

**Comment by Nobelium-XVIII** [comment]
temporarily closing this until I no longer require manual checks

**Comment by Nobelium-XVIII** [comment]
Alright there we go

**Comment by Nobelium-XVIII** [comment]
Shit right I have to make a new PR for the checks to work

It'll be back later

**Pull request closed without merging**